### ISSUE TITLE
v0.14.3 test failure on 32-bit systems

### ISSUE THREAD

**Issue by QuLogic** [PythonCharmers/python-future#149]
All [32-bit builds](https://copr.fedoraproject.org/coprs/qulogic/obspy/build/94578/) fail on 2.7 due to this test:
```
=================================== FAILURES ===================================
_______________________ TestPasteurize.test_range_slice ________________________

self = <test_future.test_pasteurize.TestPasteurize testMethod=test_range_slice>

    @skip26    # Python 2.6's lib2to3 causes the "from builtins import
               # range" line to be stuck at the bottom of the module!
    def test_range_slice(self):
        """
            After running ``pasteurize``, this Python 3 code should run
            quickly on both Py3 and Py2 without a MemoryError
            """
        code = '''
            for i in range(10**15)[:10]:
                pass
            '''
>       self.unchanged(code, from3=True)

tests/test_future/test_pasteurize.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages/future/tests/base.py:284: in unchanged
    self.convert_check(code, code, **kwargs)
../../BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages/future/tests/base.py:270: in convert_check
    from3=from3, run=run, conservative=conservative)
../../BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages/future/tests/base.py:189: in convert
    _ = self._run_test_script(interpreter=interpreter)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_future.test_pasteurize.TestPasteurize testMethod=test_range_slice>
filename = 'mytestscript.py', interpreter = '/usr/bin/python'

    def _run_test_script(self, filename='mytestscript.py',
                         interpreter=sys.executable):
        # Absolute file path:
        fn = self.tempdir + filename
        try:
            output = check_output([interpreter, fn],
                                  env=self.env, stderr=STDOUT)
        except CalledProcessError as e:
            with open(fn) as f:
                msg = (
                    'Error running the command %s\n'
                    '%s\n'
                    'Contents of file %s:\n'
                    '\n'
                    '%s') % (
                        ' '.join([interpreter, fn]),
                        'env=%s' % self.env,
                        fn,
                        '----\n%s\n----' % f.read(),
                    )
>           raise VerboseCalledProcessError(msg, e.returncode, e.cmd, output=e.output)
E           VerboseCalledProcessError: Command '['/usr/bin/python', '/tmp/tmpPTuDVU/mytestscript.py']' failed with exit status 1
E           Message: Error running the command /usr/bin/python /tmp/tmpPTuDVU/mytestscript.py
E           env={'PYTHONPATH': '/builddir/build/BUILD/future-0.14.3:/builddir/build/BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages'}
E           Contents of file /tmp/tmpPTuDVU/mytestscript.py:
E           
E           ----
E           from __future__ import unicode_literals
E           from __future__ import print_function
E           from __future__ import division
E           from __future__ import absolute_import
E           from builtins import range
E           from future import standard_library
E           standard_library.install_aliases()
E           for i in range(10**15)[:10]:
E               pass
E           
E           ----
E           Output: Traceback (most recent call last):
E             File "/tmp/tmpPTuDVU/mytestscript.py", line 8, in <module>
E               for i in range(10**15)[:10]:
E             File "/builddir/build/BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages/future/types/newrange.py", line 117, in __getitem__
E               return self.__getitem_slice(index)
E             File "/builddir/build/BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages/future/types/newrange.py", line 129, in __getitem_slice
E               start, stop, step = slce.indices(self._len)
E           OverflowError: cannot fit 'long' into an index-sized integer

../../BUILDROOT/python-future-0.14.3-1.fc21.i386/usr/lib/python2.7/site-packages/future/tests/base.py:365: VerboseCalledProcessError
== 1 failed, 866 passed, 22 skipped, 52 xfailed, 1 warnings in 123.58 seconds ==
```
I guess it's a 32-bit int problem.

**Comment by QuLogic** [comment]
The only 32-bit pass is Python 2.6.

**Comment by edschofield** [comment]
I believe this is now fixed on v0.15, although I don't have a 32-bit system to test it on. If there are still problems, please feel free to reopen this issue.

**Comment by QuLogic** [comment]
Seems to [work](https://copr.fedoraproject.org/coprs/qulogic/obspy/build/105647/).
